includes/core-changes/corefx/5.0/binaryformatter-serialization-obsolete.md

```diff
@@ -0,0 +1,92 @@
+### BinaryFormatter serialization methods are obsolete and prohibited in ASP.NET apps
+
+`Serialize` and `Deserialize` methods on <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter>, <xref:System.Runtime.Serialization.Formatter>, and <xref:System.Runtime.Serialization.IFormatter> are now obsolete. Additionally, <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter> serialization is prohibited by default for ASP.NET apps.
+
+#### Change description
+
+Due to [security vulnerabilities](../../../../docs/standard/serialization/binaryformatter-security-guide.md#binaryformatter-security-vulnerabilities) in <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter>, the following methods are now obsolete. Additionally, in ASP.NET 5.0 and later apps, they will throw a <xref:System.NotSupportedException>, unless the web app has re-enabled <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter> functionality.
+
+- <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Serialize%2A?displayProperty=nameWithType>
+- <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Deserialize%2A?displayProperty=nameWithType>
+
+These methods are marked obsolete as part of an effort to wind down usage of <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter> within the .NET ecosystem.
+
+The following serialization methods are also now obsolete, but have no behavioral changes:
+
+- <xref:System.Runtime.Serialization.Formatter.Serialize(System.IO.Stream,System.Object)?displayProperty=nameWithType>
+- <xref:System.Runtime.Serialization.Formatter.Deserialize(System.IO.Stream)?displayProperty=nameWithType>
+- <xref:System.Runtime.Serialization.IFormatter.Serialize(System.IO.Stream,System.Object)?displayProperty=nameWithType>
+- <xref:System.Runtime.Serialization.IFormatter.Deserialize(System.IO.Stream)?displayProperty=nameWithType>
+
+#### Version introduced
+
+5.0 Preview 8
+
+#### Recommended action
+
+- Stop using <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter> in your code. Instead, consider using <xref:System.Text.Json.JsonSerializer> or <xref:System.Xml.Serialization.XmlSerializer>. For more information, see [BinaryFormatter security guide](../../../../docs/standard/serialization/binaryformatter-security-guide.md).
+
+- You can temporarily suppress the <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter> compile-time warning, which is `SYSLIB0011`. We recommend that you thoroughly assess your code for risks before choosing this option. The easiest way to suppress the warnings is to surround the individual call site with `#pragma` directives.
+
+  ```csharp
+  // Now read the purchase order back from disk
+  using (FileStream readStream = new FileStream("myfile.bin", FileMode.Open))
+  {
+      BinaryFormatter formatter = new BinaryFormatter();
```

Comment:
nit

```suggestion
      var formatter = new BinaryFormatter();
```

Reply:
I guess you are suggesting this because the type is obvious from the right-hand side. @GrabYourPitchforks are you okay with this change?

Reply:
@gewarren Yes. It looks more neat to use var when the type is obvious.